modules/asg_node_group/README.md

```diff
@@ -218,7 +218,7 @@ module "bottlerocket_nodes" {
 enable the [AWS EBS CSI driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver) by setting `aws_ebs_csi_driver = true` on the cluster module.
 see: https://github.com/bottlerocket-os/bottlerocket/blob/develop/QUICKSTART-EKS.md#csi-plugin
 
-⚠️ Bottlerocket does not yet [support GPU nodes](https://github.com/bottlerocket-os/bottlerocket/issues/769), do not set `gpu = true` when `bottlerocket = true`, as this may result in an invalid configuration!
+⚠️ Bottlerocket now [supports GPU nodes](https://github.com/bottlerocket-os/bottlerocket/blob/develop/QUICKSTART-EKS.md#aws-k8s--nvidia-variants), set `gpu = true` to enable them. Ensure that you set `instance_types` to a GPU instance type.
```

Comment:
Maybe add a note about not requiring the nvidia device plugin when using bottlerocket+gpu because the AMIs come with the device plugin?

Reply:
Or update the daemonset in the module [to satisfy](https://github.com/bottlerocket-os/bottlerocket/blob/develop/QUICKSTART-EKS.md#aws-k8s--nvidia-variants):

> If you already have a daemonset for the device plugin in your cluster, you may need to use taints and tolerations to keep it from running on Bottlerocket nodes.

?

Reply:
Yeah. Good call - I suspect there's very limited use-cases for people to be running both bottlerocket and not-bottlerocket gpu nodes, but let's cater for that.

Reply:
Hmm... I think we're actually set for this, but now I'm not sure where the nvidia-device-plugin is actually supposed to run.

https://github.com/cookpad/terraform-aws-eks/blob/c39fd8600f2a67bb019ed657da5112315dfdbbf1/modules/asg_node_group/main.tf#L24

https://github.com/cookpad/terraform-aws-eks/blob/c39fd8600f2a67bb019ed657da5112315dfdbbf1/modules/cluster/addons/nvidia-device-plugin.yaml#L77-L79

Reply:
Mm I think we're not set for 

> If you already have a daemonset for the device plugin in your cluster, you may need to use taints and tolerations to keep it from running on Bottlerocket nodes.

I think with the current change we'll tag both bottlerocket and non-bottlerocket nodes with `{ "nvidia.com/gpu" = "true" }`
https://github.com/cookpad/terraform-aws-eks/blob/c39fd8600f2a67bb019ed657da5112315dfdbbf1/modules/asg_node_group/main.tf#L24

But we need to tag bottlerocket nodes with something else to prevent the nvidia device plugin scheduling there with the current toleration

Reply:
@ettiee isn't that what the `NoSchedule` effect does?

Reply:
Oh, I think I get it. It's a toleration for the corresponding taint.

Reply:
yup and we don't want a toleration for the nvidia plugin on bottlerocket gpu nodes (to solve the recommendation from bottlerocket of not scheduling the driver on bottlerocket nodes)

Reply:
hm, I suspect the simplest thing to do would be to add a further label to tag it as a bottlerocket node, and re-jig the nodeSelector.

Reply:
👍 sounds good

Reply:
I think that https://github.com/cookpad/terraform-aws-eks/pull/298/files#diff-571d8d76bc35ff080312258572db903f8fcb59a504d6d7491112722634c42a5b should do the trick.